### PR TITLE
Support eur/usd

### PR DESCRIPTION
**Pull request by tallywiesenberg**
- Adds Coinbase and OpenExchangeRate Forex APIs as PriceServices
- Adds EUR/USD DataFeed, aggregating Coinbase and OpenExchangeRate
- Adds EUR/USD to QueryCatalog

### CODE REVIEW

File: src/telliot_feeds/sources/price/currency/coinbase.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+        This implementation gets the price from the Coinbase pro API
+        using the interface described at:
+        https://docs.pro.coinbase.com/#products API
+
+        Note that the timestamp returned form the coinbase API could be used
+        instead of the locally generated timestamp.
+        """
+
+        request_url = "/v2/exchange-rates?currency={}".format(asset.upper())
```
Comment (oraclown) [inline src/telliot_feeds/sources/price/currency/coinbase.py:37]: Use f-strings instead

File: src/telliot_feeds/sources/price/currency/coinbase.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+        elif "response" in d:
+            response = d["response"]
+
+            if "message" in response:
+                logger.error(f"API ERROR ({self.name}): {response['message']}")
+                return None, None
+
+        else:
+            raise Exception("Invalid response from get_url")
```
Comment (oraclown) [inline src/telliot_feeds/sources/price/currency/coinbase.py:52]: Don't raise an error. Catch specific ones, then have one that catches everything else you can't think of (and it should log an error. ex: `logger.error("Invalid response from get_url")`

File: src/telliot_feeds/sources/price/currency/coinbase.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+        price = float(response["data"]["rates"][currency.upper()])
```
Comment (oraclown) [inline src/telliot_feeds/sources/price/currency/coinbase.py:54]: Is it possible to get a `KeyError` here?

File: src/telliot_feeds/sources/price/currency/openexchangerate.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+        This implementation gets the price from the OpenExchangeRate API
+        using the interface described at:
+        https://www.exchangerate-api.com/docs/free API
+
+        Note that the timestamp returned form the OpenExchangeRate API could be used
+        instead of the locally generated timestamp.
+        """
+
+        request_url = "/v6/latest/{}".format(asset.upper())
```
Comment (oraclown) [inline src/telliot_feeds/sources/price/currency/openexchangerate.py:37]: Use f-string instead

File: src/telliot_feeds/sources/price/currency/openexchangerate.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+        elif "response" in d:
+            response = d["response"]
+
+            if "message" in response:
+                logger.error(f"API ERROR ({self.name}): {response['message']}")
+                return None, None
+
+        else:
+            raise Exception("Invalid response from get_url")
```
Comment (oraclown) [inline src/telliot_feeds/sources/price/currency/openexchangerate.py:52]: Same thing re raising errors as above. We wan't `telliot` to keep running, but just log the error

File: src/telliot_feeds/sources/price/currency/openexchangerate.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+        price = float(response["rates"][currency.upper()])
```
Comment (oraclown) [inline src/telliot_feeds/sources/price/currency/openexchangerate.py:54]: Same thing re `KeyError` as above

File: tests/feeds/test_eur_usd_feed.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+@pytest.mark.asyncio
+async def test_fetch_price():
+    """Fetch latest EUR/USD price"""
+    (value, _) = await eur_usd_median_feed.source.fetch_new_datapoint()
+    assert value > 0
+    print(value)
+
+
+def test_query_info():
```
Comment (oraclown) [inline tests/feeds/test_eur_usd_feed.py:17]: You can remove this test. `SpotPrice` query generation of its `queryId` & `queryData` has been tested. If you want to include the `queryId` & `queryData` somewhere in the file w/ the datafeed (as a docstring or comment), then that's an option

**Review by oraclown — CHANGES_REQUESTED** [review]
Thanks @tallywiesenberg ! I left some change requests as comments

**Comment by oraclown** [comment]
> 

You need to change the function return types to match what it's telling you the function outputs

**Comment by oraclown** [comment]
> Tox typing failing on... ` src/telliot_feeds/reporters/reporter_autopay_utils.py:166: error: Returning Any from function declared to return "Optional[int]"` `src/telliot_feeds/cli/commands/report.py:397: error: Function does not return a value`

You need to change the function return types to match what it's telling you the function outputs

**Review by oraclown — APPROVED** [review]
Nice, thanks @tallywiesenberg

File: src/telliot_feeds/sources/price/currency/openexchangerate.py [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+            elif "response" in d:
+                response = d["response"]
+
+                if "message" in response:
+                    logger.error(f"API ERROR ({self.name}): {response['message']}")
+                    return None, None
+
+            else:
+                logger.error("Invalid response from get_url")
```
Comment (oraclown) [inline src/telliot_feeds/sources/price/currency/openexchangerate.py:53]: Need to return None, None here I think?

**Review by oraclown — CHANGES_REQUESTED** [review]
Left some comments and your tests are failing:
```
FAILED tests/feeds/test_eur_usd_feed.py
FAILED tests/sources/test_curr_exchange_sources.py
  ```

**Comment by tallywiesenberg** [comment]
test passing!

**Review by oraclown — APPROVED** [review]

**Pull request merged**